File: AtCoder/ABC170/D.py
N = int(input())
A = list(map(int, input().split()))
A.sort()
check = [0] * 1000001
res = 0
for i in range(N):
    if check[A[i]]:
        continue
    if i == N-1 or A[i] != A[i+1]:
        res += 1
    for j in range(A[i], 1000001, A[i]):
        check[j] = 1
print(res)

File: AtCoder/ABC170/D.rs
use proconio::input;

fn main() {
    input! {
        n : usize,
        mut a : [usize; n]
    };
    a.sort();
    let mut check = [0; 1000001];
    let mut res = 0;
    for i in 0..n {
        if check[a[i]] != 0 { continue }
        if i == n-1 || a[i] != a[i+1] { res += 1 }
        let mut idx = a[i];
        while idx <= 1000000 {
            check[idx] = 1;
            idx += a[i];
        }
    }
    println!("{}", res);
}
